### DISCUSSION TITLE
mesa-VR/

### DISCUSSION THREAD

**Discussion by giscus** [zmike/zmike.github.io#51] [Announcements]
# mesa-VR/

The New Reality

https://www.supergoodcode.com/mesa-VR/

**Comment by rhysperry111** [comment]
Hey Mike - first of all love your blog; Huge thanks for providing an entertaining way to keep up with some of the things going on in Mesa.

Would you ever consider doing not-so-serious explainers about various parts of the Linux graphics stack? I've never really found anything that is both interesting to read and not hilariously outdated (maybe you Mesa developers should just stop being so quick).

**Reply by zmike** [reply]
I'm not sure what this means. Maybe?

**Reply by deepankarsharma** [reply]
I would find it very helpful to know what debugging and developing workflows people use on the graphics stack. I am a fairly technical user and for instance currently I have a device where psr causes screen lockups, if I knew how to triage this, would love to work on a fix.

**Reply by zmike** [reply]
You'll need to be able to SSH into it in order debug and get logs when it happens.

**Reply by deepankarsharma** [reply]
Yep I can do that. I even found this, so that I could get more verbose logging - https://github.com/swaywm/wlroots/wiki/DRM-Debugging. Was looking for workflows on how devs build their own graphics stack and run it, how they debug it. Is it a matter of adding more and more tracing? Is it useful to run mesa under gdb? Maybe even a few streams of people triaging stuff on Youtube would be super helpful.

**Reply by zmike** [reply]
I'm not sure I've ever used youtube for anything programming-related unless it was watching conference presentations, so I don't have any resources to suggest here.

Mostly it's a matter of gdb attaching to stuff over ssh and maybe adding printfs to mesa.

**Reply by deepankarsharma** [reply]
Will give it a shot!

**Comment by qwertychouskie** [comment]
I fail to sense the connection to the number **3**.

**Reply by zmike** [reply]
Look harder.
